### PR TITLE
feat: show total Outside Contributors in `RossChart`

### LINKED ISSUE #3674: Feature: Add Outside Contributors to contributors chart on repo pages
### Suggested solution

In Repo Pages, we've added the contributors chart to show the evolution of internal/external contributors to a project, but we've left out the actual "ROSS Index" metric.

Can we add it to the chart next to the other metrics? We also want to use the name "Outside Contributors" instead of ROSS Index, for more clarity.

It should look something like this:

image

image

### PR DESCRIPTION
**Pull request by zeucapua**
## Description

Calculate and display the total amount of outside contributors as a stat in `RossChart`. Also renames the `new` category to `outside` in the tooltip.

## Related Tickets & Documents

Closes #3674 

## Mobile & Desktop Screenshots/Recordings
Screenshot 2024-07-05 at 10 56 49

## Steps to QA

1. Go to any repo page
2. Check the Outside Contributors stat exists
3. Calculate the total by checking the outside numbers per week via tooltip

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [x] Tier 3
- [ ] Tier 4

## [optional] What gif best describes this PR or how it makes you feel?
new-york-nyc-gif-by-sidetalk

### CODE REVIEW

**Review by nickytonline — APPROVED** [review]

**Review by brandonroberts — APPROVED** [review]

File: components/Repositories/RossChart.tsx
Comment (jpmcb) [inline components/Repositories/RossChart.tsx:67]: This copy isn't quite right: there could be _the same_ outside contributor contributing within all time buckets: i.e., if the same contributor contributed 4 times, that wouldn't be "4 outside contributors" that would be "4 outside contributions"

File: components/Repositories/RossChart.tsx
Comment (jpmcb) [inline components/Repositories/RossChart.tsx:112]: This copy change is 👍🏼 

The API probably needs an update since we end up calling the select for "internal contributors" to "new"

```ts
      .addSelect("outside_contributors.contributors", "new")
      .addSelect("returning_contributors.contributors", "recurring")
      .addSelect("org_contributors.contributors", "internal")f
```

where we look at the pr author association. Just making this note for myself since it'd be a good idea to make the API more clear in what it's doing for the ROSS

File: components/Repositories/RossChart.tsx
Comment (jpmcb) [reply components/Repositories/RossChart.tsx:67]: Hmmmm maybe the ROSS index endpoint needs a "total" section since "recurring" _also_ includes contributors with the `CONTRIBUTOR` association. This:

```ts
return stats?.contributors.reduce((prev, curr) => (prev += curr.new), 0);
```

would only count the first time outside contributors who have not contributed before (i.e., those with association "NONE")

**Comment by zeucapua** [comment]
PR returned to draft due to miscommunication. Rather than tallying the total number of new ("outside") contributors, the "Outside Contributors Index" should show the ROSS index, calculated to one number from the different weeks. There's still questions on how to display said number since they are in decimal (percentage? rounded? label based on ranges), so this PR will be on hold.

**Comment by zeucapua** [comment]
Never mind, should be total outside contributors.

**Comment by bdougie** [comment]
Screenshot 2024-07-15 at 1 32 03 PM

copy: "Outside contributors are not part of the core team managing the repository. This number represents their contributions."

cc @isabensusan or @BekahHW if this is 👍🏾 or 👎🏾

File: components/Repositories/RossChart.tsx
Comment (nickytonline) [inline components/Repositories/RossChart.tsx:118]: no need to make this a [heading](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements). Headings are typically for breaking up content/Table of Contents. If it's for styling, opt for another tag.

**Review by nickytonline — CHANGES_REQUESTED** [review]
One small tweak aside from the copy update and this is good to go.

File: components/Repositories/RossChart.tsx
Comment (zeucapua) [reply components/Repositories/RossChart.tsx:118]: It's to match the other h3 tags throughout the page as the label I've used for the stats in the different charts. Would you recommend I use just another p tag?

File: components/Repositories/RossChart.tsx
Comment (bdougie) [inline components/Repositories/RossChart.tsx]:
```suggestion
                  <InfoTooltip information="Outside contributors are not part of the core team managing the repository. This number represents their contributions." />
```

**Review by bdougie — APPROVED** [review]

File: components/Repositories/RossChart.tsx
Comment (nickytonline) [reply components/Repositories/RossChart.tsx:118]: Another tag would probably make more sense, but if it's everywhere else, for now, leave it and we can do a follow up PR.

**Review by nickytonline — APPROVED** [review]
:shipit:

**Review by jpmcb — APPROVED** [review]
Ship it 🚢 !!

**Pull request merged**
